### PR TITLE
Update release notes for 1.14

### PR DESCRIPTION
**Pull request by rnicoll**

### CODE REVIEW

**Comment by Bestbitcoinmanager** [comment]
https://t.me/DinoParkBot?start=447783279

File: doc/release-notes.md [hunk 1]
```diff
@@ -1,33 +1,45 @@
 (note: this is a temporary file, to be added-to by anybody, and moved to
 release-notes at release time)
 
-Bitcoin Core version 0.14.0 is now available from:
+Dogecoin Core version 0.14.0 is now available from:
```
Comment (langerhans) [inline doc/release-notes.md]: 1.14

File: doc/release-notes.md [hunk 2]
```diff
@@ -1,33 +1,45 @@
...
 
-  <https://bitcoin.org/bin/bitcoin-core-0.14.0/>
+  <https://dogecoin.com/bin/dogecoin-core-0.14.0/>
```
Comment (langerhans) [inline doc/release-notes.md]: That URL is probably wrong. Should point to GitHub maybe?

Comment (rnicoll) [reply doc/release-notes.md]: I've rewritten it to point to nowhere that works in Github, but mostly it's a placeholder until we get to the release itself.

**Pull request merged**